backend/src/middlewares/redisMiddleware.ts

```diff
@@ -0,0 +1,18 @@
+import { getServiceLogger } from '../utils/logging'
+import { createRedisClient } from '../utils/redis'
+
+const log = getServiceLogger()
+
+export async function redisMiddleware(req, res, next) {
+  try {
+    const redis = await createRedisClient(true)
```

Comment:
Creation of redis client is an expensive operation - the way it's usually done is that only 1 or 2 instances are created at the application boot time and they are used statically. You can still inject them here in the middleware but they shouldn't be created for each request. It is also a memory leak since you have to call `quit` method once you don't need the client anymore otherwise it keeps an open connection to redis. And here we lose the reference to the client which wasn't properly disposed of.

Reply:
Moved the redis connection init to the app boot with setImmediate :+1: middleware is getting this object and setting req.redis to this